pkg/monitor/monitor_test.go

```diff
@@ -0,0 +1,245 @@
+	driver, ok = storkdriver.(*mock.Driver)
+	require.True(t, ok, "Error casting mockdriver")
+
+	err = storkdriver.Init(nil)
+	require.NoError(t, err, "Error initializing mock volume driver")
+
+	monitor = &Monitor{
+		Driver: storkdriver,
+	}
+
+	err = monitor.Start()
+	require.NoError(t, err, "failed to start monitor")
+
+	nodes := &v1.NodeList{}
+	nodes.Items = append(nodes.Items, *newNode(nodeForPod, nodeForPod, "192.168.0.1", "rack1", "", ""))
+	nodes.Items = append(nodes.Items, *newNode("node2.domain", "node2.domain", "192.168.0.2", "rack2", "", ""))
+	nodes.Items = append(nodes.Items, *newNode("node3.domain", "node3.domain", "192.168.0.3", "rack1", "", ""))
+	nodes.Items = append(nodes.Items, *newNode("node4.domain", "node4.domain", "192.168.0.4", "rack2", "", ""))
+	nodes.Items = append(nodes.Items, *newNode("node5.domain", "node5.domain", "192.168.0.5", "rack3", "", ""))
+
+	for _, n := range nodes.Items {
+		node, err := k8s.Instance().CreateNode(&n, false)
+		require.NoError(t, err, "failed to create fake node")
+		require.NotNil(t, node, "got nil node from create node api")
+	}
+
+	provNodes := []int{0, 1}
+	err = driver.CreateCluster(5, nodes)
+	require.NoError(t, err, "Error creating cluster")
+
+	err = driver.ProvisionVolume(driverVolumeName, provNodes, 1)
+	require.NoError(t, err, "Error provisioning volume")
+}
+
+func testUnknownDriverPod(t *testing.T) {
+	pod := newPod("driverPod", []string{driverVolumeName})
+	testLostPod(t, pod, true, true, false)
+}
+
+func testUnknownOtherDriverPod(t *testing.T) {
+	pod := newPod("otherDriverPod", nil)
+	podVolume := v1.Volume{}
+	podVolume.PersistentVolumeClaim = &v1.PersistentVolumeClaimVolumeSource{
+		ClaimName: "noDriverPVC",
+	}
+	pod.Spec.Volumes = append(pod.Spec.Volumes, podVolume)
+
+	testLostPod(t, pod, false, true, false)
+}
+
+func testEvictedOtherDriverPod(t *testing.T) {
+	pod := newPod("otherDriverPod", nil)
+	podVolume := v1.Volume{}
+	podVolume.PersistentVolumeClaim = &v1.PersistentVolumeClaimVolumeSource{
+		ClaimName: "noDriverPVC",
+	}
+	pod.Spec.Volumes = append(pod.Spec.Volumes, podVolume)
+
+	testLostPod(t, pod, false, false, true)
+}
+
+func testEvictedDriverPod(t *testing.T) {
+	pod := newPod("driverPod", []string{driverVolumeName})
+	testLostPod(t, pod, true, false, true)
+}
+
+func testLostPod(
+	t *testing.T,
+	pod *v1.Pod,
+	driverPod bool,
+	testUnknownPod bool,
+	testTaintBasedEviction bool,
+) {
+	pod, err := k8s.Instance().CreatePod(pod, false)
+	require.NoError(t, err, "failed to create pod")
+	require.NotNil(t, pod, "got nil pod back from create pod")
+
+	pod, err = k8s.Instance().GetPodByName(pod.Name, "")
+	require.NoError(t, err, "failed to get pod from fake API")
+	require.NotNil(t, pod, "got nil pod back from get pod")
+
+	logrus.Infof("[debug] pod: %v", pod)
+
+	info, err := driver.GetPodVolumes(&pod.Spec, "")
+	if driverPod {
+		require.NoError(t, err, "failed to get pod from fake API")
+		require.NotNil(t, pod, "got nil pod back from get pod")
```

Comment:
Should be checking info and update error message

Reply:
fixed